src/coreclr-debug/debugConfigurationProvider.ts

```diff
@@ -0,0 +1,51 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+
+import { DotNetAttachItemsProviderFactory, AttachPicker, AttachItem } from '../features/processPicker';
+import { PlatformInformation } from '../platform';
+
+export class CoreCLRConfigurationProvider implements vscode.DebugConfigurationProvider {
+    constructor(public platformInformation: PlatformInformation) {}
+
+    public async resolveDebugConfigurationWithSubstitutedVariables(folder: vscode.WorkspaceFolder | undefined, debugConfiguration: vscode.DebugConfiguration, token?: vscode.CancellationToken): Promise<vscode.DebugConfiguration>
+    {
+        if (!debugConfiguration)
+        {
+            return null;
+        }
+
+        // Process Id is empty, handle Attach to Process Dialog.
+        if (debugConfiguration.request === "attach" && !debugConfiguration.processId)
```

Comment:
>if [](http://example.com/codeflow?start=8&length=2)

If we do NOT convert remote to use this model, do we need to ignore remote targets in the case that the user either failed to specify a process id / name, or cancelled the process selection dialog?